test/test.js

```diff
@@ -824,6 +831,45 @@ const createTestSuite = (useElectronNet) => {
       })
     })
 
+    it('should allow POST request with empty readable stream as body', function () {
+      const filepath = path.join(os.tmpdir(), 'empty-file.txt')
+      const createEmptyFile = new Promise((resolve, reject) => {
+        fs.open(filepath, 'w', (err, fd) => {
+          if (err) reject(err)
+          else fs.close(fd, resolve)
+        })
+      })
+
+      return createEmptyFile.then(() => {
+        return new Promise(resolve => {
+          const stream = fs.createReadStream(filepath)
+          stream.on('open', () => {
+            resolve(stream)
+          })
+        })
+      }).then(body => {
+        url = `${base}inspect`
+        opts = {
+          method: 'POST',
+          body,
+          useElectronNet
+        }
+        return fetch(url, opts)
+      }).then(res => {
```

Comment:
Is there a reason to test with an actual FS stream, instead of just creating a simple empty Stream manually ? It seems I can reproduce the bug with something like :

```suggestion
      const body = new stream.PassThrough().end()

      url = `${base}inspect`
      opts = {
        method: 'POST',
        body,
        useElectronNet
      }
      
      return fetch(url, opts).then(res => {
```

However, if the stream is not finished when the fetch request is processed, it seems that the error does not happen in certain cases. It's weird... Looks like a bug on electron's side, but a workaround here is indeed in order.

Reply:
(if you apply the suggestion, do not forget to remove the imports that you added, and would then become redundant)

Reply:
I used an actual FS stream for 2 reasons:
1. this is how I found the problem
2. if I remember correctly, I tried to use a `PassThrough` stream but couldn't make the test fail

If it does fail with your code, I'm happy to use this much simpler test :).

Reply:
I can confirm that, on my machine, the version I suggested does trigger the error `TypeError: Cannot read property 'done' of undefined` on electron 7 and 9 (and does not trigger any error with electron 6).

Reply:
I just merged your suggestion and removed the imports :+1: